hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenGetInfoHandler.java

```diff
@@ -58,13 +80,111 @@ public Response createEmptyResponse(@NonNull final ResponseHeader header) {
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var tokenStore = context.createStore(ReadableTokenStore.class);
+        final var op = query.tokenGetInfoOrThrow();
+        if (op.hasToken()) {
+            final var token = tokenStore.get(requireNonNull(op.token()));
+            validateFalsePreCheck(token == null || token.deleted(), INVALID_TOKEN_ID);
+        } else {
+            throw new PreCheckException(INVALID_TOKEN_ID);
+        }
     }
 
     @Override
     public Response findResponse(@NonNull final QueryContext context, @NonNull final ResponseHeader header) {
         requireNonNull(context);
         requireNonNull(header);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var config = context.configuration().getConfigData(LedgerConfig.class);
+        final var tokenStore = context.createStore(ReadableTokenStore.class);
+        final var op = query.tokenGetInfoOrThrow();
+        final var response = TokenGetInfoResponse.newBuilder();
+        final var tokenID = op.tokenOrElse(TokenID.DEFAULT);
+
+        final var responseType = op.headerOrElse(QueryHeader.DEFAULT).responseType();
+        response.header(header);
+        if (header.nodeTransactionPrecheckCode() == OK && responseType != COST_ANSWER) {
+            final var optionalInfo = infoForToken(tokenID, tokenStore, config);
+            optionalInfo.ifPresent(response::tokenInfo);
```

Comment:
we are supposed to fill the header if no token is found as I did in getccountdeetails details and receipt and record queries.

```
 final var optionalInfo = infoForAccount(
                    account, accountStore, tokensConfig, readableTokenStore, tokenRelationStore, ledgerConfig);

            if (optionalInfo.isEmpty()) {
                responseBuilder.header(header.copyBuilder()
                        .nodeTransactionPrecheckCode(FAIL_INVALID)
                        .build());
            } else {
                optionalInfo.ifPresent(responseBuilder::accountDetails);
            }
```